frontends/torch-frontend/torch-frontend/lib/Transforms/EliminateUselessOp.cpp

```diff
@@ -0,0 +1,74 @@
+//===- EliminateUselessOp.cpp -----------------------------------*- C++ -*-===//
+//
+// Copyright 2022 ByteDance Ltd. and/or its affiliates. All rights reserved.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//===----------------------------------------------------------------------===//
+
+#include "torch-frontend/Transforms/EliminateUselessOp.h"
+#include "PassDetail.h"
+
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/IR/Builders.h"
+#include "mlir/IR/BuiltinDialect.h"
+#include "mlir/IR/BuiltinOps.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+#include "torch-mlir/Dialect/Torch/IR/TorchDialect.h"
+#include "torch-mlir/Dialect/Torch/IR/TorchOps.h"
+
+using namespace mlir;
+using namespace mlir::torch;
+using namespace mlir::torch::Torch;
+
+namespace {
+// torch.profiler is useless for compilation
+struct EleminateTorchProfilerOp : public OpRewritePattern<OperatorOp> {
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(OperatorOp op,
+                                PatternRewriter &rewriter) const override {
+    llvm::StringRef name = op.getNameAttr().getValue();
+    if (name.starts_with("profiler") && op.use_empty()) {
```

Comment:
Maybe you can make this pass more general by providing a way like registration or base class?
If this pass is general enough, it might be moved under the compiler folder instead of torch-frontend.

Reply:
> Maybe you can make this pass more general by providing a way like registration or base class? 

The latest commit makes the prefix configurable. 

>  If this pass is general enough, it might be moved under the compiler folder instead of torch-frontend.

This pass is designed only for torch-frontend. We don't want these ops go into the core pipeline for `torch-mlir`.